python/mxnet/model.py

```diff
@@ -297,10 +308,10 @@ def kv_updater(index, grad, weight):
                         # pull back the sum gradients, to the same locations.
                         kv.pull(index, grad_list, priority=-index)
                 if not update_on_kvstore:
-                    opt_list = opt_state_blocks[index]
-                    # optimizea
-                    for w, g, state in zip(arg_list, grad_list, opt_list):
-                        optimizer.update(index, w, g, state)
+                    for k, p in enumerate(zip(arg_list, grad_list)):
+                        w, g = p
+                        updater(index*num_device+k, g, w)
```

Comment:
This is OK for now, But may leads to error when logic depends on index. I do prefer to use the old logic here